### DISCUSSION TITLE
Adding an Optional flag to an ingredient

### DISCUSSION THREAD

**Discussion by tjjunior** [mealie-recipes/mealie#903] [Feature Request]
_COMPLETE AND FILL THESE OUT_
- [ x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [ x] Reviewed the existing feature requests and my feature has not been requested.
- [ x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Some recipes have variations but you may not want to actually use them.  Buying for a fudge recipe and the shopping list calls for nuts, do I really need them?  Oh, I see they are tagged as optional and I don't like nuts in my fudge so I'll skip it.  For an ice cream recipe, it may give variations for strawberry, chocolate, or mint.  The shopping list ideally would provide all 3 options, but by tagging them as optional the user knows the recipe doesn't need all 3, they can pick one.  

**(Optional): Suggest A Solution**
Add a field in the recipe ingredient row with a binary Optional flag.  Allow the user to check a box when they add an ingredient if it's optional but not required.  When showing a shopping list, show the item as optional.  One consideration is if there's any benefit to providing the recipe name when showing the optional flag?  
Examples:
Nuts (Optional) vs Nuts (Optional in Grandmas Fudge Recipe)

**Additional Information**
- [x ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by hay-kot** [comment]
I'm not sure this warrants an additional feature. Why not just add `**Optional**` in the note? The notes will be copied when being added to a shopping list.
